src/main/java/net/minecraftforge/registries/tags/ITagManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.registries.tags;
+
+import net.minecraft.core.Holder;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.TagKey;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * A tag manager holds information about all tags currently bound to a forge registry.
+ * This should be preferred to any {@link Holder}-related methods.
+ */
+public interface ITagManager<V extends IForgeRegistryEntry<V>> extends Iterable<ITag<V>>
+{
+    /**
+     * Will create an empty tag if it does not exist.
+     */
+    @NotNull ITag<V> getTag(@NotNull TagKey<V> name);
+
+    @NotNull Optional<IReverseTag<V>> getReverseTag(@NotNull V value);
```

Comment:
I feel like there should be some more javadocs on this interface, especially for example on this method giving a brief explanation of what a "reverse tag" even is (or maybe better yet have `IReverseTag` and `ITag` have javadocs both on the class level and method level.